### ISSUE TITLE
Zero shap values for complex-valued neural network in Pytorch

### ISSUE THREAD

**Issue by Pranav-Goyal** [shap/shap#2177]
I'm trying to use the Shap DeepExplainer to interpret my neural network with complex-valued weights and inputs (coded in Pytorch and pytorch-complex). All my shap values are coming out to be zero, irrespective of the inputs and outputs.

I'm also getting this warning: Casting complex values to real discards the imaginary part.

Hence, I would like to know if I can use Shap for complex-valued weights and inputs in Pytorch. Will I need to modify some lines of code to be able to use it?

Thanks, in advance!

**Comment by Pranav-Goyal** [comment]
I noticed the warning was coming from line 198 of shap/explainers/_deep/deep_pytorch.py: 
phis[l][j] = (torch.from_numpy(sample_phis[l][self.data[l].shape[0]:]).to(self.device) * (X[l][j: j + 1] - self.data[l])).cpu().detach().numpy().mean(0)
I probably fixed it correctly by changing the dtype of the np.zeros arrays being appended to phis in lines 173 and 176, to np.cdouble.

However, the problem of shap values being zero still persists.
